x-pack/plugins/xpack_main/server/lib/telemetry/monitoring/get_beats_stats.js

```diff
@@ -9,48 +9,143 @@ import { createQuery } from './create_query';
 
 const HITS_SIZE = 10000; // maximum hits to receive from ES with each search
 
+const getBaseStats = () => ({
+  // stats
+  versions: {},
+  types: {},
+  outputs: {},
+  count: 0,
+  eventsPublished: 0,
+  hosts: 0,
+  // state
+  input: {
+    count: 0,
+    names: []
+  },
+  module: {
+    count: 0,
+    names: []
+  },
+});
+
+/*
+ * Update a clusters object with processed beat stats
+ * @param {Array} results - array of Beats docs from ES
+ * @param {Object} clusters - Beats stats in an object keyed by the cluster UUIDs
+ * @param {Object} clusterHostSets - the object keyed by cluster UUIDs to count the unique hosts
+ * @param {Object} clusterModuleSets - the object keyed by cluster UUIDs to count the unique modules
+ */
+export function processResults(results = [], { clusters, clusterHostSets, clusterInputSets, clusterModuleSets }) {
+  const currHits = get(results, 'hits.hits', []);
+  currHits.forEach(hit => {
+    const clusterUuid = get(hit, '_source.cluster_uuid');
+    if (clusters[clusterUuid] === undefined) {
+      clusters[clusterUuid] = getBaseStats();
+      clusterHostSets[clusterUuid] = new Set();
+      clusterInputSets[clusterUuid] = new Set();
+      clusterModuleSets[clusterUuid] = new Set();
+    }
+
+    // beats_stats
```

Comment:
I see you have this section of code here for processing `beats_stats` results and further down a section of code for processing `beats_state` results. I think it might be a bit more readable if you extracted each section into its own respective function, something like `processBeatsStatsResults` and `processBeatsStateResults`, and then call those from here.